jax/_src/lax/control_flow.py

```diff
@@ -1385,6 +1396,8 @@ def _cond_typecheck(*avals, branches, linear):
     raise core.JaxprTypeError(
       f'cond branches take input types {jaxpr0_in_avals_str}, '
       f'called with operands of type {_avals_short(op_avals)}')
+  effects = core.join_effects((b.effects for b in branches))
```

Comment:
I think we want to check that the effects are the same across branches here. That is, something like

```python
if not all(branches[0].effects == b.effects for b in branches[1:]):
  raise ...
```

Reply:
Done.